x-pack/platform/plugins/shared/actions/server/application/connector/methods/get_all/get_all.ts

```diff
@@ -111,18 +111,33 @@ async function getAllHelper({
     );
   }
 
-  const mergedResult = [
+  const inMemoryConnectorsIds: Set<string> = new Set(
+    inMemoryConnectors.map((connector) => connector.id)
+  );
+
+  const merged = await filterInferenceConnectors(esClient, [
     ...savedObjectsActions,
-    ...inMemoryConnectors.map((inMemoryConnector) => ({
-      id: inMemoryConnector.id,
-      actionTypeId: inMemoryConnector.actionTypeId,
-      name: inMemoryConnector.name,
-      isPreconfigured: inMemoryConnector.isPreconfigured,
-      isDeprecated: isConnectorDeprecated(inMemoryConnector),
-      isSystemAction: inMemoryConnector.isSystemAction,
-      ...(inMemoryConnector.exposeConfig ? { config: inMemoryConnector.config } : {}),
-    })),
-  ].sort((a, b) => a.name.localeCompare(b.name));
+    ...inMemoryConnectors,
```

Comment:
if i understand the problem correctly, this mismatch should only happen for a preconfigured inference connector where the endpoint is only configurably available in the backend and shouldn't happen for `savedObjectActions` that users have created from the UI. would it make more sense to just pass in `inMemoryConnectors` here?

Reply:
It looks like `merged` is used as the complete set of connectors in the code below this, so we have to pass `...savedObjectsAction` here, so the SO connectors show up in the result.

It does seem like we could do something like this though:

```ts  
const merged = [
    ...savedObjectsActions,
    await filterInferenceConnectors(esClient, inMemoryConnectors)
]);
```

It seems a bit harder to read than the current code though ...

I doubt there is going to be much of a perf hit with the existing code in the PR.  Customers don't have 100's of connectors, and the short-circuit for SO connectors looks like it will not be an issue.

Or maybe I'm missing something ...